sdk/servicebus/service-bus/samples/typescript/src/receiveMessagesStreaming.ts

```diff
@@ -30,26 +36,49 @@ export async function main() {
   // - See session.ts for how to receive using sessions.
   const receiver = sbClient.createReceiver(queueName);
 
-  const processMessage = async (brokeredMessage) => {
-    console.log(`Received message: ${brokeredMessage.body}`);
-    await brokeredMessage.complete();
-  };
-  const processError = async (args) => {
-    console.log(`Error from error source ${args.errorSource} occurred: `, args.error);
-  };
-
   try {
-    receiver.subscribe(
+    const subscription = receiver.subscribe(
       {
-        processMessage,
-        processError
+        processMessage: async (brokeredMessage: ServiceBusReceivedMessage) => {
+          console.log(`Received message: ${brokeredMessage.body}`);
+          await receiver.completeMessage(brokeredMessage);
+        },
+        processError: async (args: ProcessErrorArgs) => {
+          console.log(`Error from source ${args.errorSource} occurred: `, args.error);
+
+          // the handler will not stop without explicit intervention from you.
+          if (isMessagingError(args.error)) {
+            switch (args.error.code) {
```

Comment:
![image](https://user-images.githubusercontent.com/10452642/97493918-e49cdd00-1922-11eb-9b63-0a0615a5a271.png)
Do I need to cast it?

Reply:
No. That's interesting - is it possible you need to get the latest core-amqp built as well? (that's where isMessagingError comes from).

Reply:
Should we put `isMessagingError` in service-bus then?

Reply:
It's actually re-exported from our index.ts as well.

Reply:
Depending on tomorrow's discussion we might actually introduce a Service Bus specific error type, so this might get even more SB specific and not even require a re-export.

Reply:
This PR updates samples so they compile so we probably want to get this checked in earlier than that discussion. I'll come back to this with the next PR I do to address our error types.